### PR TITLE
[DOCS] reorganize pages from misc section

### PR DESCRIPTION
**Pull request by lakhanjindam**
### Why the changes in this PR are needed?

changes in the PR is related to issue #4614 
### What are the changes in this PR?

Changes in the PR are related to documentation, which involves restructuring of content and layout

### CODE REVIEW

**Review by ashutosh-narkar — COMMENTED** [review]
@lakhanjindam thanks for working on this. Just one comment on the External data section changes.

File: docs/content/external-data.md [hunk 1]
```diff
@@ -36,40 +39,152 @@ JWTs have a limited size in practice, so if your organization has too many user
 * OPA let's you check the TTL.
 * OPA has support for making HTTP requests during evaluation, which could be used to check if a JWT has been revoked.  Though if you're connecting to a remote system on every policy decision anyway, you should think about whether connecting to the authentication system directly is more appropriate (see below).
 
+## OAuth2 and OIDC Samples
```
[the remark this answers is not in the document]
Comment (lakhanjindam) [reply docs/content/external-data.md]: All Changes are done!

File: docs/content/policy-performance.md [hunk 2]
```diff
@@ -832,11 +857,13 @@ loading 10,000 rules that implement an ACL-style authorization policy consumes a
 130MB of RAM while 100,000 rules implementing the same policy (but with 10x more tuples to check)
 consumes approximately 1.1GB of RAM.
 
+OPA also provides disk storage feature to store your policies and data. Please refer to [disk storage page](../storage/#disk) for more details.
+
```
Comment (ashutosh-narkar) [inline docs/content/policy-performance.md:861]: What about something like: 

```
By default, OPA stores policy and data in-memory. OPA's disk storage feature allows policy and data to be stored on disk. See [this](../storage/#disk) for more details.
```

**Review by ashutosh-narkar — COMMENTED** [review]
@lakhanjindam this is getting close! Just one comment and also can you please squash your commits. You can find guidelines for creating commit messages [here](https://www.openpolicyagent.org/docs/latest/contrib-code/#commit-messages).

**Comment by lakhanjindam** [comment]
> @lakhanjindam this is getting close! Just one comment and also can you please squash your commits. You can find guidelines for creating commit messages [here](https://www.openpolicyagent.org/docs/latest/contrib-code/#commit-messages).

Done 👍

**Review by ashutosh-narkar — APPROVED** [review]
LGTM. Thanks @lakhanjindam for the contribution!

**Pull request merged**